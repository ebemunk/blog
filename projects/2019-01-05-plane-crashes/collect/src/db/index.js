import { Pool } from 'pg'
import { config } from 'dotenv'
import Promise from 'bluebird'

config()

const { DB_USER, DB_PASSWORD, DB_HOST, DB_PORT, DB_NAME } = process.env

const pool = new Pool({
  user: DB_USER,
  password: DB_PASSWORD,
  host: DB_HOST,
  port: DB_PORT,
  database: DB_NAME,
  Promise,
})

export function getPool() {
  return pool
}
