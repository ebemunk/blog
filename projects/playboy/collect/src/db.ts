import { Pool } from 'pg'
import Promise from 'bluebird'

import dotenv from 'dotenv'

dotenv.config()

const pool = new Pool({
  user: process.env.DB_USER,
  password: process.env.DB_PASSWORD,
  host: process.env.DB_HOST,
  port: Number(process.env.DB_PORT),
  database: process.env.DB_DATABASE,
  Promise,
})

export default function getPool() {
  return pool
}

export function insertObj(
  obj: { [key: string]: any },
  table = '"playboy"."playmates"',
) {
  const cols = Object.keys(obj).join(',')
  const valOrder = Object.keys(obj)
    .map((v, i) => `$${i + 1}`)
    .join(',')
  const sql = `
		INSERT INTO ${table}
		(${cols})
		VALUES
		(${valOrder})
	`
  return [sql, Object.values(obj)] as const
}
