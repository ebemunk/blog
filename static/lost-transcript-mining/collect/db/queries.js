export function totalLinesByChar() {
	return `
		select char_name, count(*) as lines
		from dialog
		where type='dialog' and char_name is not null
		group by char_name
		order by lines desc
	;`
}

export function allLines(char_name) {
	return `
		select string_agg(line, E'\n' order by season, episode, seq) as text
		from dialog
		where type='dialog' and char_name='${char_name}'
		group by char_name
	;`
}

export function episodeText(season, episode) {
	return `
		select string_agg(line, ' ' order by seq) as text
		from dialog
		where type='dialog' and season=${season} and episode=${episode} and line!=''
		group by season,episode
		order by season,episode
	;`
}

export function allEpisodes() {
	return `
		select distinct season, episode
		from dialog
		order by season, episode asc
	;`
}

export function episodeLengths() {
	return `
		select
			season,
			episode,
			sum(array_length(regexp_split_to_array(line, '\\s'),1)) as words,
			sum(length(line)) as chars
		from dialog
		where type='dialog'
		group by season, episode
		order by season, episode
	;`
}

export function flashbacksAndSideways() {
	return `
		select
			season,
			episode,
			array_to_json(array_agg(meta->'flashback' order by seq asc)) as flashback,
			array_to_json(array_agg(meta->'flashsideways' order by seq asc)) as flashsideways,
			array_agg(length(line)) as chars
		from dialog
		where type='dialog'
		group by season, episode
		order by season, episode
	;`
}
