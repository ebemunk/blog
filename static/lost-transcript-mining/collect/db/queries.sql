-- total lines by char
select char_name, count(*) as lines
from dialog
where type='dialog' and char_name is not null
group by char_name
order by lines desc
;

-- total lines per char per episode per season
select season, episode, char_name, count(*) as lines
from dialog
where type='dialog'
group by season, episode, char_name
order by season, episode, lines desc
;

-- char per episode per season
select distinct season, episode, char_name from dialog
where type='dialog'
order by season, episode, char_name asc
;

-- all spoken dialog prefixed with char name by episode
select season, episode, string_agg(char_name || ': ' || line, E'\n' order by seq)
from dialog
where type='dialog'
group by season,episode
order by season,episode
;

-- all dialog for each char ordered chronologically
select char_name, string_agg(line, E'\n' order by season, episode, seq)
from dialog
where type='dialog'
group by char_name
;

-- chars in every scene
select season, episode, act, scene, array_agg(distinct char_name)
from dialog
where type='dialog'
group by season, episode, act, scene
;