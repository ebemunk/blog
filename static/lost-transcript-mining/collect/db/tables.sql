-- dialog table
CREATE TABLE public.dialog
(
  id SERIAL NOT NULL,
  season integer NOT NULL,
  episode integer NOT NULL,
  seq integer NOT NULL,
  act integer NOT NULL,
  scene integer NOT NULL,
  flashback boolean NOT NULL,
  flashsideways boolean NOT NULL,
  type character varying NOT NULL,
  char_name character varying DEFAULT NULL,
  line text DEFAULT NULL,
  directions json DEFAULT '[]'::json,
  raw text DEFAULT NULL
)

-- watson personality insights
CREATE TABLE public.personality
(
  id SERIAL NOT NULL,
  char_name character varying DEFAULT NULL,
  profile json DEFAULT '{}'::json
)

-- watson tone analyzer for all episodes
CREATE TABLE public.tone
(
  season integer NOT NULL,
  episode integer NOT NULL,
  document_tone json DEFAULT '{}'::json,
  sentences_tone json DEFAULT '[]'::json
)
