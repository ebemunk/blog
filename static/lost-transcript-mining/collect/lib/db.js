export function insertObj(obj, table = 'dialog') {
	const cols = Object.keys(obj)
	.join(',')
	const valOrder = Object.keys(obj)
	.map((v, i) => `$${i+1}`)
	.join(',')
	const sql = `
		INSERT INTO ${table}
		(${cols})
		VALUES
		(${valOrder})
	`
	return [
		sql,
		Object.values(obj)
	]
}
