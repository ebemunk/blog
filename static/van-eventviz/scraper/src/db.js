import { Pool } from 'pg'
import { config } from 'dotenv'
import Promise from 'bluebird'

config()

const { DB_USER, DB_PASSWORD, DB_HOST, DB_PORT, DB_NAME } = process.env

const pool = new Pool({
  user: DB_USER,
  password: DB_PASSWORD,
  host: DB_HOST,
  port: DB_PORT,
  database: DB_NAME,
  Promise,
})

export function getPool() {
  return pool
}

export function insertObj(obj, table) {
  const cols = Object.keys(obj).join(',')
  const valOrder = Object.keys(obj)
    .map((v, i) => `$${i + 1}`)
    .join(',')
  const sql = `
		INSERT INTO ${table}
		(${cols})
		VALUES
		(${valOrder})
	`
  return [sql, Object.values(obj)]
}
