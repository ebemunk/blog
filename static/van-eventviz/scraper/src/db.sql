drop table events;
create table events (
  id character varying primary key,
  evt json not null
  details json,
  geo json
);
