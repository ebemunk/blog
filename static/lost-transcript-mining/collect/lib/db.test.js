import { insertObj } from './db'

describe('insertObj', () => {
	it('provide correct sql', () => {
		expect(insertObj({
			faf: 'etet',
			lolol: 'yaya',
			directions: [
				'some', 'things'
			]
		})).toMatchSnapshot()
	})
})
