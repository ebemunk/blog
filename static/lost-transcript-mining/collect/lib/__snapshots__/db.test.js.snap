// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insertObj provide correct sql 1`] = `
Array [
  "
		INSERT INTO dialog
		(faf,lolol,directions)
		VALUES
		($1,$2,$3)
	",
  Array [
    "etet",
    "yaya",
    Array [
      "some",
      "things",
    ],
  ],
]
`;
